ui/app/src/utils/variables.ts

```diff
@@ -36,6 +107,18 @@ export function buildGlobalVariableDefinition(variables: Variable[]): ExternalVa
     ...buildExternalVariableDefinition('global', variables),
   };
 }
+
+export function buildBuiltinVariableDefinition(variables: VariableDefinition[]): ExternalVariableDefinition {
```

Comment:
I'm not sure we want them to appear in the variables list everytime. When we edit variables.

Reply:
Same question for global var / project var. IMO, I prefer to always have them, it easily show users they exists and they are located at the end.

Reply:
There's still some differences 
1/ External variables are dynamic so user cannot guess them by experience or knowledge. While builtin variables are static and known from (future?) documentation. To me they could appear somewhere but with a different format: - Without visibility toggle since it cannot be switched on
- With a description of what is inside, and some examples 

2/ Another big difference is that we cannot override a builtin variable.

Reply:
I changed the listing design for builtin variables 😄